pkg/apis/serving/metadata_validation.go

```diff
@@ -76,3 +78,42 @@ func ValidateTimeoutSeconds(ctx context.Context, timeoutSeconds int64) *apis.Fie
 	}
 	return nil
 }
+
+// ValidateRevisionName validates name and generateName for the revisionTemplate
+func ValidateRevisionName(ctx context.Context, name, generateName string) *apis.FieldError {
+	if generateName != "" {
```

Comment:
Did we allow to use generateName before?

Reply:
If we provide valid or invalid `generateName` in service spec it will override in the `generateName` when it creates `revision`

But if at all user give invalid `generateName` something like below
```
apiVersion: serving.knative.dev/v1alpha1
kind: Service
metadata:
  name: "hello"
spec:
  template:
    metadata:
      generateName: hi@
    spec:
      containers:
       - image: savita3020/helloworld
```
user will not be notified and service will be created successfully by overriding `generateName` in revision as shown below

`kubectl get rev -n test hello-kphn8 -o yaml`

```
apiVersion: serving.knative.dev/v1alpha1
kind: Revision
metadata:
  annotations:
    serving.knative.dev/creator: kube:admin
    serving.knative.dev/lastPinned: "1565333856"
  creationTimestamp: "2019-08-09T06:57:24Z"
  generateName: hello-
  generation: 1
  labels:
    serving.knative.dev/configuration: hello
    serving.knative.dev/configurationGeneration: "1"
    serving.knative.dev/route: hello
    serving.knative.dev/service: hello
  name: hello-kphn8
```

To just to make sure i have added condition check.